app/app/components/Profile.jsx

```diff
@@ -1,31 +1,55 @@
-"use client";
-
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
+import axios from "axios";
 import Button from "./Button";
 import EditProfile from "./EditProfile";
 import styles from "./Profile.module.css";
 
-const Profile = ({ userData }) => {
+const Profile = ({ userId }) => {
+    const [userData, setUserData] = useState(null);
     const [isModalOpen, setModalOpen] = useState(false);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
+
+    useEffect(() => {
+        if (!userId) return;
 
-    if (!userData) return null;
+        const fetchUserProfile = async () => {
+            try {
+                const response = await axios.get(`http://localhost:3001/users/profile/`, {
```

Comment:
would be nice to wrap requests in some sort of wrapping function that does not require to put url on every request OR keeping url in env variable instead - or both :)